es-home/vite.config.js

```diff
@@ -0,0 +1,81 @@
+import reactRefresh from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
+import fs from "fs";
+import path from "path";
+const urlPath = process.env.URLPATH || ''
+
+function listTsFiles(options) {
+    const { srcDir, ignoredPaths = [], ignoreSubDirs = false } = options;
+    let entries = {};
+
+    function exploreDirectory(dir) {
+        const files = fs.readdirSync(dir, { withFileTypes: true });
+        files.forEach(dirent => {
+            const fullPath = path.join(dir, dirent.name);
+            if (dirent.isDirectory()) {
+                // Check if ignoring all subdirectories or specific directories
+                if (!ignoreSubDirs && !ignoredPaths.includes(dirent.name)) {
+                    exploreDirectory(fullPath);
+                }
+            } else if (dirent.name.endsWith('.ts')) {
+                const entryKey = fullPath.replace(new RegExp(`^${path.resolve(srcDir)}\/|\.ts$`, 'g'), '').replace(/\//g, '_');
+                entries[entryKey] = fullPath;
+            }
+        });
+    }
+
+    exploreDirectory(srcDir);
+    return entries;
+}
+
+export default defineConfig({
+  base: urlPath,
+  root: '.',
+  plugins: [
+    reactRefresh(),
+  ],
+  server: {
+    port: 8080,
+    strictPort: true,
+    host: '0.0.0.0',
+    base: '/',
+  },
+  build: {
+    minify: false,
```

Comment:
'minify: false' (vite does not minify code); 'preserveEntrySignatures: true' (controls how the entrypoints are exposing the exports, making sure the functions, variables, and exports are not pruned or altered)